### ISSUE TITLE
Segmentation error on arrays output to parquet files

### ISSUE THREAD

**Issue by Mickael-van-der-Beek** [duckdb/duckdb#6114]
### What happens?

Hello DuckDB team,

We are encountering an issue while testing DuckDB for the first time.

It happens when rows containing an array of integers are output to a parquet file using the DuckDB CLI.
By doing a manual binary-search, I was able to deduce that this crash depends on at least two magic numbers, the number of rows returned (`GENERATE_SERIES()` in my examples down below) and the parquet output file format `ROW_GROUP_SIZE` configuration parameter.

The same faulty query will produce inconsistent behaviours but these four have been observed:
- `Error: INTERNAL Error: Writes are not correctly aligned!?`
- `zsh: segmentation fault  ./duckdb`
- `zsh: bus error  ./duckdb`
- process hanging while using 100% of the CPU

The "magic numbers" used in the `GENERATE_SERIES()` function and the `ROW_GROUP_SIZE()` will most likely vary depending on your query and system (my guess).

### To Reproduce

On my system, this query returning `10,000` rows with one array column containing a single integer will work:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 10000) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q1.parquet' (FORMAT 'parquet');
```

If you cross this `10,000` threshold by increasing the number of rows returned to `10,001`, it will error out:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 10001) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q2.parquet' (FORMAT 'parquet');
```

If you stay at `10,000` rows but (roughly) double the size of the array column by adding an additional integer value, it will run without an issue:

```
COPY
  (
    SELECT [ g.val, g.val + ROUND(RANDOM()) ]
    FROM GENERATE_SERIES(1, 10000) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q3.parquet' (FORMAT 'parquet');
```

Running the query without the parquet export works as well:

```
SELECT [ g.val ]
FROM GENERATE_SERIES(1, 10001) AS g(val)
GROUP BY g.val
```

And exporting to a different format than parquet like CSV works too, even if you `x100` the number of rows up to `1,000,001` rows:

 ```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 1000001) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q4.csv' (FORMAT 'csv');
```

Increasing the `ROW_GROUP_SIZE` parameter of the parquet output format configuration to a number higher than the total amount of rows returned doesn't prevent the error:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 10001) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q5.parquet' (FORMAT 'parquet', ROW_GROUP_SIZE 100000);
```

However, lowering the `ROW_GROUP_SIZE` parameter value does work:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 10001) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q6.parquet' (FORMAT 'parquet', ROW_GROUP_SIZE 1228);
```

But increasing it by one from `1228` to `1229` errors out:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 10001) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q7.parquet' (FORMAT 'parquet', ROW_GROUP_SIZE 1229);
```

That fix is only temporary though and will break if you increase the amount of rows again to `16,129` rows up from `10,001`:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 16129) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q8.parquet' (FORMAT 'parquet', ROW_GROUP_SIZE 1228);
```

While one row less, so `16,128` rows, will complete successfully:

```
COPY
  (
    SELECT [ g.val ]
    FROM GENERATE_SERIES(1, 16128) AS g(val)
    GROUP BY g.val
  )
TO '/some/file-q9.parquet' (FORMAT 'parquet', ROW_GROUP_SIZE 1228);
```

### OS:

OSX ARM M1

### DuckDB Version:

N/A

### DuckDB Client:

DuckDB CLI v0.6.1

### Full Name:

Mickael van der Beek

### Affiliation:

Alliance Gravity

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the very detailed report! This looks very similar to #5779 - which was fixed a while back (#5875). The queries seem to work for me in the latest master. Could you verify that this is still broken on the development branch?

**Comment by Mickael-van-der-Beek** [comment]
@Mytherin Thank you for picking up the issue so quickly.

After having built the master branch it does indeed fix the issue!
Since I was using the latest release I thought that I would be quite close to the current commit progress.

Sorry for the wasted time and have a nice evening.
